docs/concepts.rst

```diff
@@ -0,0 +1,28 @@
+============
+Concepts
+============
+
```

Comment:
I think we should have a section "Tables" and within it, have an introduction before detailing both table types, something along the lines of:

Tables represent the location and, optionally, the column types of a SQL Database table. They are used in most Astro SDK tasks and decorators.

There are two types of tables:
1. Persistent tables
2. Temporary tables

Reply:
This is nice, I'll add it.

Reply:
This is nice, I'll add it.